### PR TITLE
fix: add missing multiprocessing.forkserver type hints

### PR DESCRIPTION
**Pull request by kkirsche**
X-Ref: https://github.com/python/typeshed/issues/6799

This adds best-effort types to `multiprocessing.forkserver` to begin to address issue https://github.com/python/typeshed/issues/6799

The source file can be viewed here:
https://github.com/python/cpython/blob/main/Lib/multiprocessing/forkserver.py

This also adds a missing argument to spawn._main as seen here:
https://github.com/python/cpython/blob/main/Lib/multiprocessing/spawn.py#L124

### CODE REVIEW

File: stdlib/multiprocessing/forkserver.pyi [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    _forkserver_pid: int | None
+    _inherited_fds: Sequence[int] | None
+    _lock: Lock
+    _preload_modules: list[str]
+
+    def __init__(self) -> None: ...
+    def _stop(self) -> None: ...
+    def _stop_unlocked(self) -> None: ...
+    def set_forkserver_preload(self, modules_names: list[str]) -> None: ...
+    def get_inherited_fds(self) -> Sequence[int] | None: ...
```
Comment (AlexWaygood) [inline stdlib/multiprocessing/forkserver.pyi:25]: For stubs, be as permissive as possible for argument annotations, and as precise as possible for return annotations. The docstring says this returns a "list of fds", and "list" is more precise than "Sequence".

```suggestion
    def get_inherited_fds(self) -> list[int] | None: ...
```

File: stdlib/multiprocessing/forkserver.pyi [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    def ensure_running(self) -> None: ...
+
+def main(
+    listener_fd: int | None,
+    alive_r: FileDescriptorLike,
+    preload: Sequence[str],
+    main_path: str | None = ...,
+    sys_path: object | None = ...,
+) -> None: ...
+def _serve_one(child_r: int, fds: Iterable[int], unused_fds: Iterable[int], handlers: Mapping[_SIGNUM, _HANDLER]) -> Any: ...
```
Comment (AlexWaygood) [inline stdlib/multiprocessing/forkserver.pyi:36]:
```suggestion
```

File: stdlib/multiprocessing/forkserver.pyi [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+def read_signed(fd: int) -> Any: ...
+def write_signed(fd: int, n: Any) -> None: ...
```
Comment (AlexWaygood) [inline stdlib/multiprocessing/forkserver.pyi:38]: `SIGNED_STRUCT` has `'q'` as the `format_value` passed to its `__init__` (https://github.com/python/cpython/blob/2833f3798dc3647e850b303a4d0fa00609a0ae9b/Lib/multiprocessing/forkserver.py#L26), and that means that only `int`s can be passed to its `.pack` method (docs here: https://docs.python.org/3/library/struct.html#format-characters)

```suggestion
def write_signed(fd: int, n: int) -> None: ...
```

**Review by AlexWaygood — CHANGES_REQUESTED** [review]

File: stdlib/multiprocessing/forkserver.pyi (hunk 3 above)
Comment (kkirsche) [reply stdlib/multiprocessing/forkserver.pyi:27]: Ah, sorry about that. Thank you for explaining how you got to that conclusion!

Comment (AlexWaygood) [reply stdlib/multiprocessing/forkserver.pyi]: > Ah, sorry about that. Thank you for explaining how you got to that conclusion!

No need to be sorry! Typing is hard :)

If the reason for something one of us is saying is every unclear, please don't be afraid to ask for clarification btw!

**Comment by AlexWaygood** [comment]
Ah, the stubtest failures on 3.11 aren't anything you need to address in this PR; it's because there's a new 3.11 beta version that's been released.

**Review by AlexWaygood — APPROVED** [review]

**Pull request merged**